core/include/kernel/interrupt.h

```diff
@@ -39,27 +60,50 @@ struct itr_chip {
 struct itr_ops {
 	void (*add)(struct itr_chip *chip, size_t it, uint32_t flags);
 	void (*enable)(struct itr_chip *chip, size_t it);
+	bool (*isenabled)(struct itr_chip *chip, size_t it);
 	void (*disable)(struct itr_chip *chip, size_t it);
+	itr_result (*rpi)(struct itr_chip *chip, size_t it);
```

Comment:
could you use more explicit labels for fields `rpi` `rsgi`and `target` ?
it makes code more readable.